### DISCUSSION TITLE
Slot not working on navbar, i need help

### DISCUSSION THREAD

**Discussion by krofax** [vuepress-theme-hope/vuepress-theme-hope#3136] [Q&A] [answered]
I'm attempting to use **GoogleTranslateSelect**, in my navbar, but it's not working, in [here](https://theme-hope.vuejs.press/demo/slot.html), you demoed a slot in **vuepress-theme-hope**, but didn't share a proper implementation like in the case of [vuepress](https://v2.vuepress.vuejs.org/reference/default-theme/extending.html#layout-slots).

**My layout**
```

<script lang="ts">
import ParentLayout from '@vuepress/theme-default/lib/client/layouts/Layout.vue'
import GoogleTranslateSelect from '@google-translate-select/vue3'

</script>
<template>
    <ParentLayout>
        <template>
            <div class="footer-col">
                    <GoogleTranslateSelect #navbarStartBefore default-language-code="en" default-page-language-code="en"
                        :fetch-browser-language="false" trigger="click" />
                </div>
        </template>
    </ParentLayout>
</template>
```

**My client.ts**
```

import { defineClientConfig } from "@vuepress/client";

import LanguageLayout from "./layouts/LanguageDropdown.vue";

export default defineClientConfig({
    layouts: { LanguageLayout }
});

```
**My config**
```

alias: {
    '@theme-hope/layouts/LanguageDropdown': path.resolve(
      dirname,
      './components/LanguageDropdown.vue',
    ),
  },
```

**Comment by Mister-Hope** [comment, marked as answer]
1. Use correct syntax to append codes in your post.
2. Read and understand Vue slots carefully. https://vuejs.org/guide/components/slots.html#scoped-slots

**Reply by krofax** [reply]
@Mister-Hope I've used the correct syntax.

**Reply by Mister-Hope** [reply]
#navbarStartBefore should be on template.

It's definitely not my job to teach you vue basics. Again, read official docs carefully, and watch some tutorial video if you still do not get how vue works.
